webapi/webapi-sensors-w3c-tests/sensors/resources/generic-sensor-tests.js

```diff
@@ -141,9 +137,60 @@ function runGenericSensorOnerror(sensorType, sensorName) {
     let sensor = new sensorType();
     sensor.onactivate = t.step_func_done(assert_unreached);
     sensor.onerror = t.step_func_done(event => {
+      assert_equals(sensor.reading, null);
       assert_equals(sensor.state, 'errored');
       assert_equals(event.error.name, 'NotFoundError');
     });
     sensor.start();
   }, "Test that 'onerror' event is fired when " + sensorName + " sensor is not supported");
 }
+
+function runSensorFrequency(sensorType) {
+  test(() => {
+    assert_throws(new RangeError(), () => new sensorType({frequency: -60}))
+  }, "Test that negative frequency causes exception from constructor.");
+
+  async_test(t => {
+    let sensor1 = new sensorType({frequency: 30});
+    let sensor2 = new sensorType({frequency: 10});
+    sensor1.start();
+    sensor2.start();
+    let number1 = 0;
+    let number2 = 0;
+    sensor1.onchange = () => {
+      number1++;
+    };
+    sensor2.onchange = () => {
+      number2++;
+    };
+    sensor1.onerror = t.step_func_done(event => {
+      assert_unreached(event.error.name + ":" + event.error.message);
+    });
+    sensor2.onerror = t.step_func_done(event => {
+      assert_unreached(event.error.name + ":" + event.error.message);
+    });
+    t.step_timeout(() => {
+      assert_equals(Math.round(number1/number2), 3);
+      sensor1.stop();
+      sensor2.stop();
+      t.done();
+    }, 1000);
+  }, "Test that the frequency hint is correct.");
+
+  async_test(t => {
+    let sensor = new sensorType({frequency: 600});
+    sensor.start();
```

Comment:
I don't think above two tests are well designed, what's the test result of them?